ui/app/AppLayouts/Wallet/controls/FollowingAddressMenu.qml

```diff
@@ -0,0 +1,140 @@
+        root.address = model.address;
+        root.ensName = model.ensName;
+        root.tags = model.tags;
+        popup(parent, x, y);
+    }
+
+    onClosed: {
+        root.name = "";
+        root.address = "";
+        root.ensName = ""
+        root.tags = []
+    }
+
+    StatusSuccessAction {
+        id: copyAddressAction
+        objectName: "copyFollowingAddressAction"
+        successText: qsTr("Address copied")
+        text: qsTr("Copy address")
+        icon.name: "copy"
+        timeout: 1500
+        autoDismissMenu: true
+        onTriggered: ClipboardUtils.setText(d.visibleAddress)
+    }
+
+    StatusAction {
+        text: qsTr("Show address QR")
+        objectName: "showQrFollowingAddressAction"
+        assetSettings.name: "qr"
+        onTriggered: {
+            Global.openShowQRPopup({
+                                       showSingleAccount: true,
+                                       showForSavedAddress: false,
+                                       switchingAccounsEnabled: false,
+                                       hasFloatingButtons: false,
+                                       name: root.name,
+                                       address: root.address
+                                   })
+        }
+    }
+
+    StatusAction {
+        text: qsTr("View activity")
+        objectName: "viewActivityFollowingAddressAction"
+        assetSettings.name: "wallet"
+        onTriggered: {
+            Global.changeAppSectionBySectionType(Constants.appSection.wallet,
+                                                 WalletLayout.LeftPanelSelection.AllAddresses,
+                                                 WalletLayout.RightPanelSelection.Activity,
+                                                 {savedAddress: root.address})
+        }
+    }
+
+    StatusMenuSeparator {}
+
+    BlockchainExplorersMenu {
+        id: blockchainExplorersMenu
+        flatNetworks: root.activeNetworks
+        onNetworkClicked: {
+            let link = Utils.getUrlForAddressOnNetwork(shortname, isTestnet, d.visibleAddress ? d.visibleAddress : root.ensName);
+            Global.openLinkWithConfirmation(link, StatusQUtils.StringUtils.extractDomainFromLink(link));
+        }
+    }
+
+    StatusMenuSeparator { }
+
+    StatusAction {
+        text: {
+            var savedAddr = WalletStores.RootStore.getSavedAddress(root.address)
+            var isSaved = savedAddr && savedAddr.address !== ""
+            return isSaved ? qsTr("Already in saved addresses") : qsTr("Add to saved addresses")
+        }
+        assetSettings.name: {
+            var savedAddr = WalletStores.RootStore.getSavedAddress(root.address)
+            var isSaved = savedAddr && savedAddr.address !== ""
+            return isSaved ? "star-icon" : "star-icon-outline"
+        }
+        objectName: "addToSavedAddressesAction"
+        enabled: {
+            var savedAddr = WalletStores.RootStore.getSavedAddress(root.address)
```

Comment:
This `WalletStore.RootStore` is a singleton right now (probably the last one) and will be refactored soon. Because of that, please take this store via public API of that that component (and others as well) and inside the component use it as a regular, non-singleton object.